deploy/zfs-operator.yaml

```diff
@@ -1205,1423 +1073,980 @@ status:
 subjects:
-  - kind: ServiceAccount
-    name: openebs-zfs-controller-sa
-    namespace: kube-system
+- kind: ServiceAccount
+  name: openebs-zfs-controller-sa
+  namespace: openebs
 roleRef:
   kind: ClusterRole
   name: openebs-zfs-snapshotter-role
   apiGroup: rbac.authorization.k8s.io
-
----
-
-########################################
-###########                 ############
-###########   Node plugin   ############
-###########                 ############
-########################################
-
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: openebs-zfs-node-sa
-  namespace: kube-system
-
----
-
-kind: ClusterRole
-apiVersion: rbac.authorization.k8s.io/v1
-metadata:
-  name: openebs-zfs-driver-registrar-role
-rules:
-  - apiGroups: [""]
-    resources: ["events"]
-    verbs: ["get", "list", "watch", "create", "update", "patch"]
-  - apiGroups: [""]
-    resources: ["persistentvolumes", "nodes", "services"]
-    verbs: ["get", "list"]
-  - apiGroups: ["*"]
-    resources: ["zfsvolumes", "zfssnapshots", "zfsbackups", "zfsrestores", "zfsnodes"]
-    verbs: ["get", "list", "watch", "create", "update", "patch"]
-
 ---
-
+# Source: zfs-localpv/templates/rbac.yaml
 kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: openebs-zfs-driver-registrar-binding
+  labels:
+    chart: zfs-localpv-2.7.0-develop
+    heritage: Helm
+    openebs.io/version: "2.7.0-develop"
+    role: "openebs-zfs"
+    name: "openebs-zfs-node"
+    release: openebs
+    openebs.io/component-name: "openebs-zfs-node"
 subjects:
-  - kind: ServiceAccount
-    name: openebs-zfs-node-sa
-    namespace: kube-system
+- kind: ServiceAccount
+  name: openebs-zfs-node-sa
+  namespace: openebs
 roleRef:
   kind: ClusterRole
   name: openebs-zfs-driver-registrar-role
   apiGroup: rbac.authorization.k8s.io
-
----
-
-kind: ConfigMap
-apiVersion: v1
-metadata:
-  name: openebs-zfspv-bin
-  namespace: kube-system # should be the same namespace where it is getting mounted
-data:
-  zfs: |
-    #!/bin/sh
-    if [ -x /host/sbin/zfs ]; then
-      chroot /host /sbin/zfs "$@"
-    elif [ -x /host/usr/sbin/zfs ]; then
-      chroot /host /usr/sbin/zfs "$@"
-    else
-      chroot /host zfs "$@"
-    fi
-
----
-
-apiVersion: scheduling.k8s.io/v1
-kind: PriorityClass
-metadata:
-  name: openebs-zfs-csi-node-critical
-# A PriorityClass object can have any 32-bit integer value smaller than or equal
-# to 1 billion. Larger numbers are reserved for critical system Pods that should
-# not normally be preempted or evicted. The higher the value, the higher the priority.
-value: 900001000
-globalDefault: false
-description: "This priority class should be used for the OpenEBS ZFS localPV CSI driver node deployment only."
-
 ---
-
+# Source: zfs-localpv/templates/zfs-node.yaml
 kind: DaemonSet
 apiVersion: apps/v1
 metadata:
-  name: openebs-zfs-node
-  namespace: kube-system
+  name: openebs-zfs-localpv-node
```

Comment:
labels and namespace seem to be different?

Reply:
The labels are fine, I have reverted the namespace. Tested the upgrade. It seems like we have two new apps, i.e controller deploy and node ds, because the names have changed. The older ones need to be removed.

Reply:
Changes:
1. Made namespace `kube-system` as it was.
2. Made the analytics installer type configurable.